storage/chunk_data_packs.go

```diff
@@ -9,7 +13,7 @@ type ChunkDataPacks interface {
 
 	// Store stores multiple ChunkDataPacks cs keyed by their ChunkIDs in a batch.
 	// No errors are expected during normal operation, but it may return generic error
```

Comment:
For every function taking a storage lock, please make sure you document the lock diligently (ideally with some context). 

```suggestion
	// StoreByChunkID persists multiple [flow.ChunkDataPack] keyed by their ChunkIDs in a batch.
	//
	// CAUTION:
	//   - The chunk ID is _not_ a collision resistant hash of the chunk data pack. For any given chunk, an Execution Node should
	//     always provide the _same_ chunk data pack (slashable protocol violation otherwise). To prevent bugs in higher-lever logic
	//     from accidentally overwriting an existing chunk data pack with a different one, we must atomically read and write data here.
	//     Hence, the caller must acquire the [storage.LockInsertChunkDataPack] and hold it until the database write has been committed.
	//
	// No error returns are expected during normal operation.
```

Reply:
This method is refactored in https://github.com/onflow/flow-go/pull/7983/files